panel-app/src/components/store.ts

```diff
@@ -130,22 +106,34 @@ function updateState(newState): boolean {
 				metadata &&
 				Object.keys(metadata).reduce((acc, key) => {
 					const item = metadata[key];
-					const itemName = Object.keys(item)[0];
-					const { type, path, isBoolean } = Object.values(item)[0] as { type: string; path: string; isBoolean: boolean };
-					acc[getId()] = { name: itemName, type: type, path: path, ...(isBoolean && { isBoolean: isBoolean }) };
+					const { type, path, isBoolean } = item;
+					acc[getId()] = { name: key, type: type, path: path, ...(isBoolean && { isBoolean: isBoolean }) };
 					return acc;
 				}, {});
 
 			const formattedSubItems =
 				subItems &&
 				Object.keys(subItems).map((key) => {
 					const { exclude, metadata } = getSubItemValues(parsedValue, key);
+					const formattedSubItemExclude =
+						exclude &&
+						exclude.map((item) => {
+							return { id: getId(), type: item.type, path: item.path };
+						});
+					const formattedSubItemMetadata =
```

Comment:
I see this logic with `reduce` a few times in this file. Can we have a function `getFormattedMetadata()`?
They are all the same, no?
2/4